src/server/saved_objects/saved_objects_mixin.js

```diff
@@ -0,0 +1,30 @@
+import { SavedObjectsClient } from './client';
+
+import {
+  createCreateRoute,
+  createDeleteRoute,
+  createFindRoute,
+  createReadRoute,
+  createUpdateRoute
+} from './routes';
+
+export function savedObjectsMixin(kbnServer, server) {
+  const prereqs = {
+    getSavedObjectsClient: {
+      assign: 'savedObjectsClient',
+      method(request, reply) {
+        reply(new SavedObjectsClient(
+          server.config().get('kibana.index'),
+          request,
```

Comment:
Just curious, why couple the `SavedObjectsClient` to the request object? The `request` object is something that it going to change dramatically over time (especially as we move to the new platform), and I don't *think* there is any benefit to exposing it to the `SavedObjectsClient` unless truly necessary. Perhaps you've identified a benefit that I'm missing?